### PR TITLE
fix(#9914): Add upload file max size limit

### PR DESCRIPTION
**Pull request by pengyejun**
### SUMMARY
CSV Upload - add max filesize limit.
To slove https://github.com/apache/incubator-superset/issues/9914

### TEST PLAN
CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -326,6 +326,7 @@ def _try_json_readsha(filepath, length):  # pylint: disable=unused-argument
 # The file upload folder, when using models with files
 UPLOAD_FOLDER = BASE_DIR + "/app/static/uploads/"
 UPLOAD_CHUNK_SIZE = 4096
+UPLOAD_MAX_SIZE = 50 * 1024 * 1024
```
Comment (mistercrunch) [inline superset/config.py:329]: NIT: how about `UPLOAD_MAX_BYTES` to clarify the unit

File: superset/exceptions.py [hunk 2]
```diff
@@ -71,6 +71,10 @@ class CertificateException(SupersetException):
     message = _("Invalid certificate")
 
 
+class UploadOverFlowException(SupersetException):
```
Comment (mistercrunch) [inline superset/exceptions.py:74]: NIT: `UploadOverflowException` : lower case `f`

File: superset/views/database/views.py [hunk 3]
```diff
@@ -65,9 +65,15 @@ def certificate_form_validator(_, field: StringField) -> None:
...
 def upload_stream_write(form_file_field: "FileStorage", path: str):
     chunk_size = app.config["UPLOAD_CHUNK_SIZE"]
+    max_size = app.config["UPLOAD_MAX_SIZE"] or float("inf")
     with open(path, "bw") as file_description:
         while True:
             chunk = form_file_field.stream.read(chunk_size)
+            max_size -= len(chunk)
+            if max_size < 0:
+                raise UploadOverFlowException(
+                    _("Upload Max Size %(size)s", size=app.config["UPLOAD_MAX_SIZE"])
```
Comment (mistercrunch) [inline superset/views/database/views.py:75]: NIT: "Exceeded the upload maximum size of %(size)s bytes"

File: superset/config.py (hunk 1 above)
Comment (pengyejun) [reply superset/config.py]: Thank you very much for your suggestion, it has been adopted

**Review by tooptoop4 — APPROVED** [review]

File: superset/views/database/views.py [hunk 4]
```diff
@@ -65,9 +65,18 @@ def certificate_form_validator(_, field: StringField) -> None:
 
 def upload_stream_write(form_file_field: "FileStorage", path: str):
     chunk_size = app.config["UPLOAD_CHUNK_SIZE"]
+    max_size = app.config["UPLOAD_MAX_BYTES"] or float("inf")
     with open(path, "bw") as file_description:
         while True:
             chunk = form_file_field.stream.read(chunk_size)
+            max_size -= len(chunk)
```
Comment (bkyryliuk) [inline superset/views/database/views.py:72]: please add the unit test

**Pull request closed without merging**